src/rgw/rgw_redis_driver.cc

```diff
@@ -0,0 +1,682 @@
+#include <boost/algorithm/string.hpp>
+#include "rgw_redis_driver.h"
+//#include "rgw_ssd_driver.h" // fix -Sam
+#include <boost/asio/experimental/awaitable_operators.hpp>
+#include <boost/redis/src.hpp>
+
+#define dout_subsys ceph_subsys_rgw
+#define dout_context g_ceph_context
+
+namespace rgw { namespace cache { 
+
+std::unordered_map<std::string, Partition> RedisDriver::partitions;
+
+std::vector< std::pair<std::string, std::string> > build_attrs(rgw::sal::Attrs* binary) 
+{
+  std::vector< std::pair<std::string, std::string> > values;
+  rgw::sal::Attrs::iterator attrs;
+
+  /* Convert to vector */
+  if (binary != NULL) {
+    for (attrs = binary->begin(); attrs != binary->end(); ++attrs) {
+      values.push_back(std::make_pair(attrs->first, attrs->second.to_str()));
+    }
+  }
+
+  return values;
+}
+
+int RedisDriver::find_client(const DoutPrefixProvider* dpp) 
+{
+  if (client.is_connected())
+    return 0;
+
+  if (addr.host == "" || addr.port == 0) { 
+    ldpp_dout(dpp, 10) << "RGW Redis Cache: Redis cache endpoint was not configured correctly" << dendl;
+    return EDESTADDRREQ;
```

Comment:
please return *negative* integer error codes to be consistent with the rest of rgw